pkg/driver/options.go

```diff
@@ -137,11 +144,17 @@ func (o *Options) Validate() error {
 	if o.MetricsCertFile != "" || o.MetricsKeyFile != "" {
 		switch {
 		case o.HTTPEndpoint == "":
-			return errors.New("--http-endpoint MUST be specififed when using the metrics server with HTTPS")
+			return errors.New("--http-endpoint MUST be specified when using the metrics server with HTTPS")
 		case o.MetricsCertFile == "":
-			return errors.New("--metrics-cert-file MUST be specififed when using the metrics server with HTTPS")
+			return errors.New("--metrics-cert-file MUST be specified when using the metrics server with HTTPS")
 		case o.MetricsKeyFile == "":
-			return errors.New("--metrics-key-file MUST be specififed when using the metrics server with HTTPS")
+			return errors.New("--metrics-key-file MUST be specified when using the metrics server with HTTPS")
+		}
+	}
+
+	for _, source := range strings.Split(o.MetadataSources, ",") {
+		if source != metadata.SourceIMDS && source != metadata.SourceK8s {
+			return metadata.InvalidSourceErr(strings.Split(o.MetadataSources, ","), source)
```

Comment:
If we use a StringSliceVar, it would be ideal to normalize and validate once here:

```
for i, s := range o.MetadataSources {
    s = strings.ToLower(strings.TrimSpace(s))
    switch s {
    case metadata.SourceIMDS, metadata.SourceK8s:
        o.MetadataSources[i] = s
    default:
        return metadata.InvalidSourceErr(o.MetadataSources, s)
    }
}

```

passing the slice downstream makes it so that `strings.Split` isn't needed in `main.go` again.